packages/wonder-blocks-dropdown/components/dropdown-core.js

```diff
@@ -64,78 +79,167 @@ export default class DropdownCore extends React.Component<DropdownCoreProps> {
+            // right away, it returns the value of the document width instead
+            // of the width of the just-mounted component, the menu.
+            requestAnimationFrame(() => {
+                // $FlowFixMe
+                const rect = this.menuRef.getBoundingClientRect();
+                // This update forcing would only happen once, because now
+                // this.menuWidth is set
+                this.menuWidth = rect.width;
+                this.forceUpdate();
+            });
+        }
     }
 
     componentWillUnmount() {
-        document.removeEventListener("mouseup", this._handleInteract);
-        document.removeEventListener("touchend", this._handleInteract);
+        this.removeEventListeners();
+    }
+
+    updateEventListeners() {
+        if (this.props.open) {
+            this.addEventListeners();
+        } else {
+            this.removeEventListeners();
+        }
+    }
+
+    addEventListeners() {
+        // This is click instead of mouse up because events happen in this order
+        // mousedown -> mouseup -> click for mouse presses. We want the event
+        // to be handled by the onClick callback on the menu first before
+        // attempting to handle this event (which would close the menu). The
+        // menu callback stops propagation of the event.
+        document.addEventListener("click", this._handleInteract);
+        document.addEventListener("touch", this._handleInteract);
+        // This is still keyup because we are listening for the escape key,
+        // which invokes events keydown and keyup and NOT keypress.
+        document.addEventListener("keyup", this._handleKeyup);
+    }
+
+    removeEventListeners() {
+        document.removeEventListener("click", this._handleInteract);
+        document.removeEventListener("touch", this._handleInteract);
         document.removeEventListener("keyup", this._handleKeyup);
     }
 
     _handleInteract = (event: {target: any}) => {
-        if (this.node && this.node.contains(event.target)) {
-            return;
-        }
         if (this.props.open) {
-            this.props.onClose();
+            this.props.onDropdownChanged(false);
         }
     };
 
     _handleKeyup = (event: KeyboardEvent) => {
         if (this.props.open && event.key === "Escape") {
             event.preventDefault();
             event.stopImmediatePropagation();
-            this.props.onClose();
+            this.props.onDropdownChanged(false);
+        }
+    };
+
+    updatePosition = () => {
+        const {openerNode} = this.props;
+        if (openerNode) {
+            //$FlowFixMe typing is weird for the return value of ref callback
+            const rect = openerNode.getBoundingClientRect();
+            this.openerBounds = {
+                top: rect.top + window.pageYOffset,
+                left: rect.left + window.pageXOffset,
+                width: rect.width,
+            };
```

Comment:
`openerBounds` is only used in a single place, we could just return it and use the value there instead of storing it on `this`.